mikeio/dfsu.py

```diff
@@ -1110,42 +1175,76 @@ def plot(
+                
             else:
-                # must be contourf plot then
-                levels = np.linspace(vmin, vmax, n_levels)
-                tr_fig = ax.tricontourf(triang, zn, levels=levels, cmap=cmap)
-                # if plot_type == 'contour_lines':
-                #     ax.tricontour(triang, zn, levels=levels,
-                #             colors=['0.5'],
-                #             linewidths=[0.5])
-
-            plt.colorbar(tr_fig, label=label)
-
-        ax.set_xlim(nc[:, 0].min(), nc[:, 0].max())
-        ax.set_ylim(nc[:, 1].min(), nc[:, 1].max())
+                if (plot_type is not None) and plot_type != 'outline_only':
+                    raise Exception(f'plot_type {plot_type} unknown!')
 
-    def _create_tri_only_element_table(self, geometry=None):
+            if show_mesh and (not geometry.is_tri_only):
+                # if mesh is not tri only, we need to add it manually on top
+                patches = geometry._to_polygons()
+                mesh_linewidth = 0.4
+                if plot_type == 'contour': mesh_col = mesh_col_dark
+                p = PatchCollection(patches, edgecolor=mesh_col, facecolor='none', linewidths=mesh_linewidth)
+                ax.add_collection(p)    
+           
+        if show_outline:
+            mp = self.to_shapely()
+            domain = mp.buffer(0)
+            out_col = "0.4"
+            ax.plot(*domain.exterior.xy, color=out_col, linewidth=1.2)
+            for j in range(len(domain.interiors)):
+                interj = domain.interiors[j]
+                ax.plot(*interj.xy, color=out_col, linewidth=1.2)
+
+        if title is not None:
+            ax.set_title(title)
+
+        return fig_obj
+
+    def _create_tri_only_element_table(self, data=None, geometry=None):
         """Convert quad/tri mesh to pure tri-mesh
         """
         if geometry is None:
             geometry = self
 
         ec = geometry.element_coordinates
         if geometry.is_tri_only:
-            return np.asarray(geometry.element_table), ec
+            return np.asarray(geometry.element_table), ec, data 
+        
+        if data is None:
+            data = []
 
-        elem_table = [
-            list(geometry.element_table[i]) for i in range(geometry.n_elements)
-        ]
+        elem_table = [list(geometry.element_table[i]) for i in range(geometry.n_elements)]
         tmp_elmnt_nodes = elem_table.copy()
         for el, item in enumerate(tmp_elmnt_nodes):
             if len(item) == 4:
                 elem_table.pop(el)
-                elem_table.insert(el, item[:3])
-                elem_table.append([item[i] for i in [2, 3, 0]])
-                ec = np.append(ec, ec[el].reshape(1, -1), axis=0)
+                elem_table.insert(el,item[:3])
+                elem_table.append([item[i] for i in [2,3,0]])
+                data = np.append(data, data[el])
+                ec = np.append(ec, ec[el].reshape(1, -1), axis=0)    # THIS IS NOT CORRECT! 
```

Comment:
Interesting comment

Reply:
I don't understand why spaces have been removed after comas. The  standard in Python is to put space after each language element. The only difference from that standarf that I know of is when defining default parameters in a functions' header when  the '=' operator is linked to the variable name and its default value. Ex: variable='10' instead of variable = '10'

Reply:
me neither - will add back in

Reply:
yes, forgot all about that calculation of ec - I will fix that

Reply:
done

Reply:
> I don't understand why spaces have been removed after comas. The standard in Python is to put space after each language element. The only difference from that standard that I know of is when defining default parameters in a functions' header when the '=' operator is linked to the variable name and its default value. Ex: variable='10' instead of variable = '10'

We rely on [Black](https://black.readthedocs.io/en/stable/) for formatting code, which can be used to auto-format on save, so you don't have to remember the rules of a style guide.

Reply:
I use PyCharm for Python work and it complained about formatting not conforming to PEP 8.  PyCharm also verifies and rearranges the code. If you use PyCharm, Black would not be necessary. 

If you look at the lines I mentioned, there are inconsistencies, some commas have spaces after them some don't. This means that either Black failed or has not been used for the file. 
Also, I noticed that Black is in "beta"